src/streamlink/plugins/atresplayer.py

```diff
@@ -23,30 +22,18 @@
     r"https?://(?:www\.)?atresplayer\.com/",
 ))
 class AtresPlayer(Plugin):
+    _channels_api_url = "https://api.atresplayer.com/client/v1/info/channels"
+    _player_api_url = "https://api.atresplayer.com/player/v1/live/{channel_id}"
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.url = update_scheme("https://", f"{self.url.rstrip('/')}/")
 
     def _get_streams(self):
-        path = urlparse(self.url).path
-        api_url = self.session.http.get(self.url, schema=validate.Schema(
-            re.compile(r"""window.__PRELOADED_STATE__\s*=\s*({.*?});""", re.DOTALL),
-            validate.none_or_all(
-                validate.get(1),
-                validate.parse_json(),
-                {"links": {path: {"href": validate.url()}}},
-                validate.get(("links", path, "href")),
-            ),
-        ))
-        if not api_url:
-            return
-        log.debug(f"API URL: {api_url}")
-
-        player_api_url = self.session.http.get(api_url, schema=validate.Schema(
-            validate.parse_json(),
-            {"urlVideo": validate.url()},
-            validate.get("urlVideo"),
-        ))
+        channel_path = re.search(r"(\/\w+\/)$", self.url).group()
```

Comment:
Avoid unnecessary regexes

```suggestion
        channel_path = f"/{self.url.split('/')[-2]}/"
```

Reply:
This however won't capture the channel_path `"/"`, which is one of the items in the API response. Yours won't either. Since I can't access the site and its streams, I don't know which channels this belongs to.

Reply:
My change are meant work with urls like: https://www.atresplayer.com/directos/CHANNEL which are live streams, eg: https://www.atresplayer.com/directos/lasexta - which are also the urls that are in the tests. In that context, channel "/" would be undefined.

We could make the URL matching more strict (to only match /directos/) but I'm not sure is worth...